x/ccv/provider/types/msg.go

```diff
@@ -83,167 +71,215 @@ func NewMsgAssignConsumerKey(consumerId string, providerValidatorAddress sdk.Val
 	}, nil
 }
 
-// Route implements the sdk.Msg interface.
-func (msg MsgAssignConsumerKey) Route() string { return RouterKey }
-
-// Type implements the sdk.Msg interface.
-func (msg MsgAssignConsumerKey) Type() string {
-	return TypeMsgAssignConsumerKey
-}
-
-// GetSigners implements the sdk.Msg interface. It returns the address(es) that
-// must sign over msg.GetSignBytes().
-// If the validator address is not same as delegator's, then the validator must
-// sign the msg as well.
-func (msg MsgAssignConsumerKey) GetSigners() []sdk.AccAddress {
-	valAddr, err := sdk.ValAddressFromBech32(msg.ProviderAddr)
-	if err != nil {
-		// same behavior as in cosmos-sdk
-		panic(err)
+// ValidateBasic implements the sdk.HasValidateBasic interface.
+func (msg MsgAssignConsumerKey) ValidateBasic() error {
+	if err := validateDeprecatedChainId(msg.ChainId); err != nil {
+		return errorsmod.Wrapf(ErrInvalidMsgAssignConsumerKey, "ChainId: %s", err.Error())
 	}
-	return []sdk.AccAddress{valAddr.Bytes()}
-}
-
-// GetSignBytes returns the message bytes to sign over.
-func (msg MsgAssignConsumerKey) GetSignBytes() []byte {
-	bz := ccvtypes.ModuleCdc.MustMarshalJSON(&msg)
-	return sdk.MustSortJSON(bz)
-}
 
-// ValidateBasic implements the sdk.Msg interface.
-func (msg MsgAssignConsumerKey) ValidateBasic() error {
 	if err := ValidateConsumerId(msg.ConsumerId); err != nil {
-		return err
+		return errorsmod.Wrapf(ErrInvalidMsgAssignConsumerKey, "ConsumerId: %s", err.Error())
 	}
+
 	_, err := sdk.ValAddressFromBech32(msg.ProviderAddr)
 	if err != nil {
-		return ErrInvalidProviderAddress
+		return errorsmod.Wrapf(ErrInvalidMsgAssignConsumerKey, "ProviderAddr: %s", err.Error())
```

Comment:
ditto: the error messages don't contain the expected argument.

Reply:
I added a validateProviderAddress method.